engine/src/main/java/org/terasology/persistence/serializers/EntityDataJSONFormat.java

```diff
@@ -80,7 +80,7 @@ public static String write(EntityData.Entity entity) {
         try {
             return newGson().fromJson(reader, EntityData.Prefab.class);
         } catch (JsonSyntaxException e) {
-            throw new IOException("Failed to load prefab", e);
+            return null;
```

Comment:
Same here - the JsonSyntaxExcpetion is caught, and the signature indicates that an IOException is thrown in case of an error.